### DISCUSSION TITLE
How to Upgrade App using operateRelease API in kubeapps?

### DISCUSSION THREAD

**Discussion by sehrishnaseer** [kubeapps/kubeapps#4002] [Q&A] [answered]
I am trying to upgrade my release, both through UI and operateRelease API, in kubeops section, but getting the following response. I am getting the response that I should give auth.password, but where am I supposed to do that? Both in API and in UI. Any help regarding this issue will be highly appreciated!
![ss1](https://user-images.githubusercontent.com/95912089/147539455-bd313fbf-8c6c-458d-b8f2-19b63aed3ca1.png)

![Ss](https://user-images.githubusercontent.com/95912089/147539228-0a78a0e3-535a-48fc-9348-e52db1240f8c.png)

**Comment by absoludity** [comment, marked as answer]
Hi there @sehrishnaseer . The error you're seeing there is related to the chart's expectation about how credentials for that chart are handled. We shouldn't be returning a 500 there, but rather a 400 (created #4022 for that).

Most bitnami charts will allow you to deploy an app without you specifying a secret for an admin password, creating the admin password for you during the initial deploy, but then requiring you to explicitly specify that password to helm (via the values or CLI) during an upgrade. For more info, see

https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases

You can avoid this behavior by explicitly creating a secret manually before the initial deploy and referring to that in your values.

**Reply by sehrishnaseer** [reply]
I have one more question, that can a Statefulset storage can be upgraded? Because while doing an upgrade using UpgradeInstalledPackage API, I keep getting an error that Stafefulset can not be upgraded, although Deployment's storage can be upgraded using the API. Is there any possible way out, to upgrade the Statefulset's storage?

**Reply by absoludity** [reply]
Kubeapps shouldn't be doing anything here that is different to upgrading the app via the helm CLI (assuming this is a Helm chart that you're playing with). Can you reproduce the same issue via the Helm CLI or do you see a different result? If so, if you can provide a demo chart with which I can reproduce the issue, I can dig deeper to find out what Kubeapps is doing differently from the Helm CLI. Thanks!
